rent_ai/admin.py

```
@@ -0,0 +1,116 @@
+    @action(
+        description="Sync from external source",
+        url_path="sync-from-external-source",
+        permissions=["sync_from_external_source"],
+    )
+    def sync_from_external_source(self, request: HttpRequest, object_id: int):
+        setting = Setting.get_solo()
+        try:
+            result = setting.update_from_external_source()
+            level = messages.ERROR if result.startswith("Failed") or result.startswith("Invalid") else messages.SUCCESS
+            self.message_user(request, result, level=level)
+        except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)